### PR TITLE
fix(autoheal): Heal multiple partitions

### PR DESCRIPTION
**Pull request by ieQu1**

### CODE REVIEW

File: test/mria_autoheal_SUITE.erl [hunk 1]
```diff
@@ -81,41 +67,25 @@ t_autoheal_with_replicants(Config) when is_list(Config) ->
        try
            Nodes = [N1, N2, N3, N4, N5] = mria_ct:start_cluster(mria, Cluster),
            ok = mria_mnesia_test_util:wait_tables(Nodes),
-           %% Simulate netsplit
-           true = rpc:cast(N1, net_kernel, disconnect, [N2]),
-           true = rpc:cast(N1, net_kernel, disconnect, [N3]),
-           ok = timer:sleep(1000),
-           %% SplitView: {[N2,N3], [N1]}
-           assert_partition_contents(N1, #{running => [N1], stopped => [N2, N3]}),
```
Comment (ieQu1) [inline test/mria_autoheal_SUITE.erl:89]: This part duplicated the test done in `t_autoheal` and it appeared unstable, so I removed it.

File: src/mria_autoheal.erl [hunk 2]
```diff
@@ -79,13 +80,21 @@ handle_msg(Msg = {create_splitview, Node}, Autoheal = #autoheal{delay = Delay, t
...
-                {_Views, BadNodes} ->
-                    ?LOG(critical, "Bad nodes found when autoheal: ~p", [BadNodes])
-            end,
+            RPCResult = erpc:multicall(Nodes, mria_mnesia, running_nodes, []),
+            SplitView = lists:foldl(fun({N, Result}, Acc) ->
+                                            case Result of
+                                                {ok, Peers} ->
+                                                    Acc #{N => Peers};
+                                                _ ->
+                                                    %% Ignore unreachable nodes:
```
Comment (ieQu1) [inline src/mria_autoheal.erl:89]: This changes the behavior, but the old behavior was strange: when multicall failed, the autoheal procedure was aborted and never restarted.

File: test/mria_autoheal_SUITE.erl [hunk 3]
```diff
@@ -200,26 +170,27 @@ assert_replicant_bootstrapped(R, C, Trace) ->
...
+    {Trace, _} = ?split_trace_at(#{?snk_kind := teardown_cluster}, Trace0),
+    {_, [HealEvent|AfterHeal]} = ?split_trace_at(#{?snk_kind := "Rebooting minority"}, Trace),
+    #{nodes := Minority} = HealEvent,
+    [?assert(
+        ?strict_causality( #{?snk_kind := mria_exec_callback, type := stop, ?snk_meta := #{node := N}}
+                         , #{?snk_kind := mria_exec_callback, type := start, ?snk_meta := #{node := N}}
+                         , AfterHeal
+                         ))
+     || N <- Minority],
+    true.
```
Comment (thalesmg) [inline test/mria_autoheal_SUITE.erl:184]: Q: would it be relevant to assert that the majority didn't restart?

**Review by thalesmg — APPROVED** [review]

File: test/mria_autoheal_SUITE.erl (hunk 3 above)
Comment (ieQu1) [reply test/mria_autoheal_SUITE.erl:184]: Good idea, I will add such check.

Comment (ieQu1) [reply test/mria_autoheal_SUITE.erl:189]: Done.

**Review by thalesmg — APPROVED** [review]

**Review by SergeTupchiy — APPROVED** [review]

**Pull request merged**